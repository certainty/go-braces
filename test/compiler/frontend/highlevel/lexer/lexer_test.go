package lexer_test

import (
	"path/filepath"
	"testing"

	"github.com/certainty/go-braces/pkg/compiler/frontend/highlevel/lexer"
	"github.com/certainty/go-braces/pkg/compiler/frontend/highlevel/token"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/assert"
)

var fixtureDir = "../../../../testdata/compiler"

func TestMinimal(t *testing.T) {
	source, err := lexer.NewFileInput(filepath.Join(fixtureDir, "minimal.brace"))
	assert.NoError(t, err)

	scanner := lexer.New(source)
	tokens := readTokens(scanner)
	assert.Empty(t, scanner.Errors)

	snaps.MatchSnapshot(t, tokens)
}

func readTokens(scanner *lexer.Scanner) []token.Token {
	tokens := []token.Token{}

	for {
		tok := scanner.NextToken()
		tokens = append(tokens, tok)
		if tok.Type == token.EOF {
			break
		}
	}

	return tokens
}
