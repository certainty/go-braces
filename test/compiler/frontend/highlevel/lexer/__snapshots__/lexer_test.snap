
[TestMinimal - 1]
(tok PROC "proc" (loc 3 (2 6)))
(tok IDENTIFIER "main" (loc 3 (7 11)))
(tok LPAREN "(" (loc 3 (11 12)))
(tok RBRACE ")" (loc 3 (12 13)))
(tok LBRACE "{" (loc 3 (14 15)))
(tok FIXNUM "3" (loc 4 (18 19)))
(tok ADD "+" (loc 4 (20 21)))
(tok FIXNUM "5" (loc 4 (22 23)))
(tok MUL "*" (loc 4 (24 25)))
(tok FIXNUM "4" (loc 4 (26 27)))
(tok RBRACE "}" (loc 5 (28 29)))
(tok EOF "" (loc 6 (30 30)))
---
